https://github.com/checkstyle/checkstyle/actions/runs/25450602815/job/74666425294?pr=19689#step:5:418

` com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception was thrown while processing ./.ci-temp/jdk25/test/jdk/jdk/jfr/event/tracing/TestConstructors.java`

looks like unrelated problem, but we need to find solution for it.

https://github.com/openjdk/jdk/blob/master/test/jdk/jdk/jfr/event/compiler/TestCompilerCompile.java

parsing of this file is fine: https://github.com/checkstyle/checkstyle/actions/runs/25602984979/job/75159952356


https://github.com/checkstyle/checkstyle/actions/runs/25450602818/job/74666424968?pr=19689#step:6:191
performance run failing with same Input file:
```
================ MOST RECENT COMMAND RESULT =================
Starting audit...
line 27:12 token recognition error at: '\'
line 27:22 token recognition error at: '\'
line 66:38 token recognition error at: '\'
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception was thrown while processing ./.ci-temp/jdk25/test/jdk/jdk/jfr/event/tracing/TestConstructors.java
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:313)
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:227)
	at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:429)
	at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:347)
	at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:205)
	at com.puppycrawl.tools.checkstyle.Main.main(Main.java:129)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: IllegalStateException occurred while parsing file /home/runner/work/checkstyle/checkstyle/./.ci-temp/jdk25/test/jdk/jdk/jfr/event/tracing/TestConstructors.java.
	at com.puppycrawl.tools.checkstyle.JavaParser.parse(JavaParser.java:104)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:186)
	at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:101)
	at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:341)
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:300)
	... 5 more
Caused by: java.lang.IllegalStateException: 54:17: mismatched input '(' expecting ';'
	at com.puppycrawl.tools.checkstyle.JavaParser$CheckstyleErrorListener.syntaxError(JavaParser.java:254)
	at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41)
	at org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:544)
	at org.antlr.v4.runtime.DefaultErrorStrategy.reportInputMismatch(DefaultErrorStrategy.java:327)
	at org.antlr.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:139)
	at com.puppycrawl.tools.checkstyle.CheckstyleParserErrorStrategy.recoverInline(CheckstyleParserErrorStrategy.java:38)
	at org.antlr.v4.runtime.Parser.match(Parser.java:208)
	at com.puppycrawl.tools.checkstyle.grammar.java.JavaLanguageParser.statement(JavaLanguageParser.java:7064)
	at com.puppycrawl.tools.checkstyle.grammar.java.JavaLanguageParser.blockStatement(JavaLanguageParser.java:6225)
	at com.puppycrawl.tools.checkstyle.grammar.java.JavaLanguageParser.constructorBlock(JavaLanguageParser.java:5938)
	at com.puppycrawl.tools.checkstyle.grammar.java.JavaLanguageParser.constructorDeclaration(JavaLanguageParser.java:3088)
	at com.puppycrawl.tools.checkstyle.grammar.java.JavaLanguageParser.memberDeclaration(JavaLanguageParser.java:2778)
	at com.puppycrawl.tools.checkstyle.grammar.java.JavaLanguageParser.classBodyDeclaration(JavaLanguageParser.java:2690)
	at com.puppycrawl.tools.checkstyle.grammar.java.JavaLanguageParser.classBody(JavaLanguageParser.java:2496)
	at com.puppycrawl.tools.checkstyle.grammar.java.JavaLanguageParser.classDeclaration(JavaLanguageParser.java:1115)
	at com.puppycrawl.tools.checkstyle.grammar.java.JavaLanguageParser.memberDeclaration(JavaLanguageParser.java:2799)
	at com.puppycrawl.tools.checkstyle.grammar.java.JavaLanguageParser.classBodyDeclaration(JavaLanguageParser.java:2690)
	at com.puppycrawl.tools.checkstyle.grammar.java.JavaLanguageParser.classBody(JavaLanguageParser.java:2496)
	at com.puppycrawl.tools.checkstyle.grammar.java.JavaLanguageParser.classDeclaration(JavaLanguageParser.java:1115)
	at com.puppycrawl.tools.checkstyle.grammar.java.JavaLanguageParser.types(JavaLanguageParser.java:772)
	at com.puppycrawl.tools.checkstyle.grammar.java.JavaLanguageParser.typeDeclaration(JavaLanguageParser.java:686)
	at com.puppycrawl.tools.checkstyle.grammar.java.JavaLanguageParser.compilationUnit(JavaLanguageParser.java:420)
	at com.puppycrawl.tools.checkstyle.JavaParser.parse(JavaParser.java:98)
	... 9 more
Caused by: org.antlr.v4.runtime.InputMismatchException
	... 27 more
Checkstyle ends with 1 errors.

```

--- a/config/checker-framework-suppressions/checker-nullness-optional-interning-suppressions.xml
+++ b/config/checker-framework-suppressions/checker-nullness-optional-interning-suppressions.xml
@@ -4820,17 +4820,6 @@
   <checkerFrameworkError unstable="false">
     <fileName>src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/ClassDefHandler.java</fileName>
     <specifier>argument</specifier>
-    <message>incompatible argument for parameter ast of AbstractExpressionHandler.getLineStart.</message>
-    <lineContent>final int lineStart = getLineStart(ident);</lineContent>
-    <details>
-      found   : @Initialized @Nullable DetailAST
-      required: @Initialized @NonNull DetailAST
-    </details>
-  </checkerFrameworkError>
-
-  <checkerFrameworkError unstable="false">
-    <fileName>src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/ClassDefHandler.java</fileName>
-    <specifier>argument</specifier>
     <message>incompatible argument for parameter ast of AbstractExpressionHandler.isOnStartOfLine.</message>
     <lineContent>if (isOnStartOfLine(atAst)) {</lineContent>
     <details>
@@ -4851,6 +4840,13 @@
     <specifier>dereference.of.nullable</specifier>
     <message>dereference of possibly-null reference getMainAst().findFirstToken(TokenTypes.OBJBLOCK)</message>
     <lineContent>return getMainAst().findFirstToken(TokenTypes.OBJBLOCK)</lineContent>
+  </checkerFrameworkError>
+
+  <checkerFrameworkError unstable="false">
+    <fileName>src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/ClassDefHandler.java</fileName>
+    <specifier>dereference.of.nullable</specifier>
+    <message>dereference of possibly-null reference ident</message>
+    <lineContent>if (ident.getLineNo() == getMainAst().getLineNo()) {</lineContent>
   </checkerFrameworkError>
 
   <checkerFrameworkError unstable="false">

--- a/config/checkstyle-examples-suppressions.xml
+++ b/config/checkstyle-examples-suppressions.xml
@@ -70,9 +70,6 @@
   <!-- Example deals with line wrapping and check does not take that into account.  -->
   <suppress id="exampleClassName"
     files="nolinewrap[\\/]Example5.java"/>
-  <!-- until https://github.com/checkstyle/checkstyle/issues/2629 -->
-  <suppress checks="Indent"
-    files="nolinewrap[\\/]Example5.java"/>
 
   <!-- multifileregexpheader examples use external XML configs instead of inline -->
   <suppress id="exampleFileXmlConfigPresence"

--- a/config/projects-to-test/openjdk25-excluded.files
+++ b/config/projects-to-test/openjdk25-excluded.files
@@ -742,6 +742,9 @@
 <module name="BeforeExecutionExclusionFileFilter">
   <property name="fileNamePattern" value="[\\/]test[\\/]langtools[\\/]tools[\\/]javac[\\/]SuperInit[\\/]EarlyInnerAccessErrorMessageTest.java$"/>
 </module>
+<module name="BeforeExecutionExclusionFileFilter">
+  <property name="fileNamePattern" value="[\\/]test[\\/]jdk[\\/]jdk[\\/]jfr[\\/]event[\\/]tracing[\\/]TestConstructors.java$"/>
+</module>
 <!-- Parenthesized patterns removed from grammar -->
 <module name="BeforeExecutionExclusionFileFilter">
   <property name="fileNamePattern" value="[\\/]test[\\/]langtools[\\/]tools[\\/]javac[\\/]patterns[\\/]Parenthesized.java$"/>

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/ClassDefHandler.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/ClassDefHandler.java
@@ -74,9 +74,13 @@ public class ClassDefHandler extends BlockParentHandler {
         else {
             if (getMainAst().getType() != TokenTypes.ANNOTATION_DEF) {
                 final DetailAST ident = getMainAst().findFirstToken(TokenTypes.IDENT);
-                final int lineStart = getLineStart(ident);
+                DetailAST tokenToCheck = getMainAst();
+                if (ident.getLineNo() == getMainAst().getLineNo()) {
+                    tokenToCheck = ident;
+                }
+                final int lineStart = getLineStart(tokenToCheck);
                 if (!getIndent().isAcceptable(lineStart)) {
-                    logError(ident, "ident", lineStart);
+                    logError(tokenToCheck, "ident", lineStart);
                 }
             }
         }

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
@@ -4282,6 +4282,22 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
             getPath("InputIndentationNewWithTernaryOp.java"), expected);
     }
 
+    @Test
+    public void testClassDefIndentation() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
+
+        checkConfig.addProperty("tabWidth", "4");
+        checkConfig.addProperty("basicOffset", "4");
+        checkConfig.addProperty("braceAdjustment", "0");
+        checkConfig.addProperty("caseIndent", "4");
+        checkConfig.addProperty("throwsIndent", "8");
+        checkConfig.addProperty("forceStrictCondition", "false");
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        verifyWarns(checkConfig,
+                getPath("InputIndentationClassDef.java"),
+                expected);
+    }
+
     /**
      * Test to kill Pitest mutation by verifying internal state is cleared.
      *

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationClassDef.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationClassDef.java
@@ -1,0 +1,18 @@
+/* Config:                                                                   //indent:0 exp:0
+ *                                                                           //indent:1 exp:1
+ * basicOffset = 4                                                           //indent:1 exp:1
+ * braceAdjustment = 0                                                       //indent:1 exp:1
+ * caseIndent = 4                                                            //indent:1 exp:1
+ * forceStrictCondition = false                                              //indent:1 exp:1
+ * lineWrappingIndentation = 4                                               //indent:1 exp:1
+ * throwsIndent = 8                                                          //indent:1 exp:1
+ */                                                                          //indent:1 exp:1
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;      //indent:0 exp:0
+
+public                                                                       //indent:0 exp:0
+    class InputIndentationClassDef {                                         //indent:4 exp:4
+}                                                                            //indent:0 exp:0
+
+class                                                                        //indent:0 exp:0
+    Test1 {                                                                  //indent:4 exp:4
+}                                                                            //indent:0 exp:0

